File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.cpp
class MyQueue {
private:
    stack<int> one;
    stack<int> two;
    
public:
    MyQueue() {
        
    }
    
    void push(int x) {
        while (!one.empty()) {
            two.push(one.top());
            one.pop();
        }
        one.push(x);
        while (!two.empty()) {
            one.push(two.top());
            two.pop();
        }
    }
    
    int pop() {
        int res = one.top();
        one.pop();
        return res;
    }
    
    int peek() {
        return one.top();
    }
    
    bool empty() {
        return one.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.java
class MyQueue {

    private Stack<Integer> one;
    private Stack<Integer> two;
    
    public MyQueue() {
        this.one = new Stack<>();
        this.two = new Stack<>();
    }
    
    public void push(int x) {
        one.push(x);
    }
    
    public int pop() {
        while (one.size() != 1) {
            two.push(one.pop());
        }
        int res = one.pop();
        while (!two.isEmpty()) {
            one.push(two.pop());
        }
        return res;
    }
    
    public int peek() {
        // System.out.println(one);
        while (one.size() != 1) {
            two.push(one.pop());
        }
        // System.out.println(one);
        int res = one.pop();
        one.push(res);
        while (!two.isEmpty()) {
            one.push(two.pop());
        }
        return res;
    }
    
    public boolean empty() {
        return one.isEmpty();
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */